breadth_first_search/breadth_first_search.py

```diff
@@ -0,0 +1,92 @@
+"""
+Breadth-first search (BFS) is an algorithm for traversing or searching tree or
+graph data structures. Starting at the tree root (or some arbitrary node of a
+graph, sometimes referred to as a 'search key'[1]) and explores the neighbor
+nodes first, before moving to the next level neighbours.
+"""
+
+from collections import deque
+
+
+def breadth_first_search(graph, source):
```

Comment:
@SteadBytes: The breadth first **search** algorithm should be used for searching the path from one node to another in the graph in a breadth first manner.

In this implementation, you are trying to list out the breadth first traversal path, while you should be searching for a node. 

Please let me know if something is wrong with this explanation.

Reply:
That is a very good point! I guess I was trying to show how a breadth first search would traverse the structure during the search - which makes this a breadth first traversal algorithm as apposed to search.

It will be easy enough to fix though, should I fix and make a new PR?

Reply:
Please update this one itself. Thanks :)

Reply:
Ok I will do, I just noticed there is a directory for `depth_first_traversal`, I could move this algorithm into a `breadth_first_traversal` directory and then make the changes to this for `breadth_first_search`.

Would that be ok?

Reply:
Adding breadth first traversal can go to another PR. Or you can close this PR, create another PR with different change in README and make it as breadth first traversal.

I prefer the first option.

Reply:
Ok sounds good, I will do that - thanks a lot :)